Objects/unicodeobject.c

```diff
@@ -8998,28 +9012,10 @@ PyUnicode_Count(PyObject *str,
             goto onError;
     }
 
-    switch (kind1) {
-    case PyUnicode_1BYTE_KIND:
-        result = ucs1lib_count(
-            ((const Py_UCS1*)buf1) + start, end - start,
-            buf2, len2, PY_SSIZE_T_MAX
-            );
-        break;
-    case PyUnicode_2BYTE_KIND:
-        result = ucs2lib_count(
-            ((const Py_UCS2*)buf1) + start, end - start,
-            buf2, len2, PY_SSIZE_T_MAX
-            );
-        break;
-    case PyUnicode_4BYTE_KIND:
-        result = ucs4lib_count(
-            ((const Py_UCS4*)buf1) + start, end - start,
-            buf2, len2, PY_SSIZE_T_MAX
-            );
-        break;
-    default:
-        Py_UNREACHABLE();
-    }
+    result = anylib_count(kind1,
+                          str, buf1 + start, end - start,
```

Comment:
Can you please double check that we don't need to use any casts here? I don't quite understand what `(const Py_UCS1*)buf1)` was used for.